lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Runtime/HostManager/HostManager.h"
+#include "glow/Backends/DeviceManager.h"
+#include "glow/Graph/Context.h"
+#include "glow/Partitioner/Partitioner.h"
+#include "glow/Runtime/Provisioner/Provisioner.h"
+#include "glow/Runtime/RuntimeTypes.h"
+// #include "glow/Runtime/Executor.h"
+
+#include <future>
+#include <queue>
+
+using namespace glow;
+using namespace runtime;
+
+HostManager::HostManager(const std::vector<DeviceConfig> &configs) {
+  DeviceIDTy deviceCount = 0;
+  for (auto &config : configs) {
+    // auto newDevice = DeviceManager.getDevice(config);
+    // devices_.emplace(deviceCount, newDevice);
+    deviceCount++;
+  }
+  // executor_ = Executor(devices_);
+}
+
+HostManager::~HostManager() { clearHost(); }
+
+ResultCode HostManager::addNetwork(Module *M) {
+  std::vector<DeviceInfo> deviceInfo;
+  for (auto &device : devices_) {
+    DeviceInfo info = DeviceInfo();
+    info.availableMemory = device.second->getAvailableMemory();
```

Comment:
should not `getAvailableMemory` get synchronized?
when do we actually place network on a device?

Reply:
The HostManager doesn't keep track of device memory, it's likely the DeviceManager will keep it in sync, I don't envision this call to query the device each time.